llvm/test/CodeGen/DirectX/metadata-stripping.ll

```diff
@@ -24,9 +24,9 @@ _Z4mainDv3_j.exit:                                ; preds = %for.body.i, %entry
 ; These next check lines check that only the range metadata remains
 ; No more metadata should be necessary, the rest (the current 0 and 1)
 ; should be removed.
-; CHECK-NOT: !{!"llvm.loop.mustprogress"}
-; CHECK: [[RANGEMD]] = !{i32 1, i32 5}
-; CHECK-NOT: !{!"llvm.loop.mustprogress"}
+; CHECK-DAG: [[RANGEMD]] = !{i32 1, i32 5}
+; CHECK-DAG: [[LOOPMD]] = distinct !{[[LOOPMD]], [[PROGRESS:![0-9]+]]}
+; CHECK-DAG: {!"llvm.loop.mustprogress"}
```

Comment:
```suggestion
; CHECK-DAG: [[LOOPMD]] = distinct !{[[LOOPMD]], [[PROGRESS:![0-9]+]]}
; CHECK-DAG: [[PROGRESS]] = {!"llvm.loop.mustprogress"}
```
Since you named `PROGRESS` you might as well use it. Or do not name it at all.